libsolidity/formal/ModelChecker.cpp

```diff
@@ -34,16 +36,51 @@ ModelChecker::ModelChecker(
 	ReadCallback::Callback const& _smtCallback,
 	smtutil::SMTSolverChoice _enabledSolvers
 ):
+	m_errorReporter(_errorReporter),
 	m_settings(_settings),
 	m_context(),
 	m_bmc(m_context, _errorReporter, _smtlib2Responses, _smtCallback, _enabledSolvers, m_settings),
 	m_chc(m_context, _errorReporter, _smtlib2Responses, _smtCallback, _enabledSolvers, m_settings)
 {
 }
 
+// TODO This should be removed for 0.9.0.
+void ModelChecker::searchPragma(vector<shared_ptr<SourceUnit>> const& _sources)
+{
+	bool hasPragma = ranges::any_of(_sources, [](auto _source) {
+		return _source ? _source->annotation().experimentalFeatures.count(ExperimentalFeature::SMTChecker) : false;
```

Comment:
Not sure why, but the ternary operator feels a bit harder to read than just this:

```suggestion
		return _source && _source->annotation().experimentalFeatures.count(ExperimentalFeature::SMTChecker);
```